### PR TITLE
Fix labels multiple insertion

### PR DESCRIPTION
**Pull request by BenjaminDeleze**
Fix https://github.com/cyberbotics/wrestling/issues/18

### CODE REVIEW

**Comment by BenjaminDeleze** [comment]
Fix it directly when receiving the update message from the supervisor.

Also fix a bug where the `.json` was broken when there was a `\n` in the text message.

File: src/webots/nodes/utils/WbSupervisorUtilities.cpp [hunk 1]
```diff
@@ -681,26 +681,40 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
...
 
+      int labelId;
       if (id < MAX_LABELS) {
-        int labelId = (int)id + mRobot->uniqueId() * MAX_LABELS;  // kind of hack to avoid an id clash.
-
-        mLabelIds.removeAll(labelId);
-        mLabelIds << labelId;
+        labelId = (int)id + mRobot->uniqueId() * MAX_LABELS;  // kind of hack to avoid an id clash.
+      } else {
+        mRobot->warn(tr("wb_supervisor_set_label() is out of range. The supported range is [0, %1].").arg(MAX_LABELS));
```
Comment (omichel) [inline src/webots/nodes/utils/WbSupervisorUtilities.cpp:688]:
```suggestion
        mRobot->warn(tr("wb_supervisor_set_label() is out of range. The supported range is [0, %1].").arg(MAX_LABELS - 1));
```

File: src/webots/nodes/utils/WbSupervisorUtilities.cpp [hunk 2]
```diff
@@ -681,26 +681,40 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
...
-        emit labelChanged(createLabelUpdateString(label));
-      } else
-        mRobot->warn(tr("wb_supervisor_set_label() is out of range. The supported range is [0, %1].").arg(MAX_LABELS));
+      }
+
+      mLabelIds.removeAll(labelId);
+      mLabelIds << labelId;
+
+      WbWrenLabelOverlay *label = WbWrenLabelOverlay::createOrRetrieve(labelId, filename);
+      QString error = label->getFontError();
```
Comment (omichel) [inline src/webots/nodes/utils/WbSupervisorUtilities.cpp:707]:
```suggestion
      const QString error = label->getFontError();
```

File: src/webots/nodes/utils/WbSupervisorUtilities.cpp [hunk 3]
```diff
@@ -681,26 +681,40 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
...
       }
 
+      int labelId;
       if (id < MAX_LABELS) {
-        int labelId = (int)id + mRobot->uniqueId() * MAX_LABELS;  // kind of hack to avoid an id clash.
-
-        mLabelIds.removeAll(labelId);
-        mLabelIds << labelId;
+        labelId = (int)id + mRobot->uniqueId() * MAX_LABELS;  // kind of hack to avoid an id clash.
+      } else {
```
Comment (omichel) [inline src/webots/nodes/utils/WbSupervisorUtilities.cpp:687]:
```suggestion
      else {
```

File: src/webots/nodes/utils/WbSupervisorUtilities.cpp [hunk 4]
```diff
@@ -681,26 +681,40 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
         filename = WbStandardPaths::fontsPath() + "Arial.ttf";
       }
 
+      int labelId;
       if (id < MAX_LABELS) {
```
Comment (omichel) [inline src/webots/nodes/utils/WbSupervisorUtilities.cpp:685]:
```suggestion
      if (id < MAX_LABELS)
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**